sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -2834,6 +2835,29 @@ class DataFrameSuite extends QueryTest
       df10.select(zip_with(col("array1"), col("array2"), (b1, b2) => reverseThenConcat2(b1, b2)))
     checkAnswer(test10, Row(Array(Row("cbaihg"), Row("fedlkj"))) :: Nil)
   }
+
+  test("SPARK-34882: Aggregate with multiple distinct null sensitive aggregators") {
+    spark.udf.register("countNulls", udaf(new Aggregator[JLong, JLong, JLong] {
```

Comment:
We need this udaf when adding this test in not `SQLQueryTestSuite` but `DataFrameSuite`?

Reply:
nit: wrap `withUserDefinedFunction`?

Reply:
I added the `withUserDefinedFunction`, but I did not understand the first question.
I added this udaf, because the built in aggregates, that are "null sensitive" (`First` and `Last`) gave unstable test results in the `SQLQueryTestSuite`.